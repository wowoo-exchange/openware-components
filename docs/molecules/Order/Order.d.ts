import * as React from 'react';
export declare type FormType = 'buy' | 'sell';
export declare type DropdownElem = number | string | React.ReactNode;
export interface OrderProps {
    type: FormType;
    orderType: string | React.ReactNode;
    price: number | string;
    amount: number | string;
}
export declare type OnSubmitCallback = (order: OrderProps) => void;
export interface OrderComponentProps {
    /**
     * Amount of money in base currency wallet
     */
    availableBase: number;
    /**
     * Amount of money in quote currency wallet
     */
    availableQuote: number;
    /**
     * Payment fee that is used in total price calculation
     */
    feeBuy: number;
    /**
     * Payment fee that is used in total price calculation
     */
    feeSell: number;
    /**
     * Callback which is called when a form is submitted
     */
    onSubmit: OnSubmitCallback;
    /**
     * If orderType is 'Market' this value will be used as price for buy tab
     */
    priceMarketBuy: number;
    /**
     * If orderType is 'Market' this value will be used as price for sell tab
     */
    priceMarketSell: number;
    /**
     * If orderType is 'Limit' this value will be used as price
     */
    priceLimit?: number;
    /**
     * Name of currency for price field
     */
    from: string;
    /**
     * Name of currency for amount field
     */
    to: string;
    /**
     * Whether order is disabled to execute
     */
    disabled?: boolean;
    handleSendType?: (index: number, label: string) => void;
    /**
     * Index of tab to switch on
     */
    tabIndex?: number;
    /**
     * Precision of amount, total, available, fee value
     */
    currentMarketAskPrecision: number;
    /**
     * Precision of price value
     */
    currentMarketBidPrecision: number;
    /**
     * @default 'Order Type'
     * Text for order type dropdown label.
     */
    orderTypeText?: string;
    /**
     * @default 'Price'
     * Text for Price field Text.
     */
    priceText?: string;
    /**
     * @default 'Amount'
     * Text for Amount field Text.
     */
    amountText?: string;
    /**
     * @default 'Total'
     * Text for Total field Text.
     */
    totalText?: string;
    /**
     * @default 'Available'
     * Text for Available field Text.
     */
    availableText?: string;
    /**
     * @default 'Estimated fee'
     * Text for Estimated fee field Text.
     */
    estimatedFeeText?: string;
    /**
     * @default 'BUY'
     * Text for buy order submit button.
     */
    submitBuyButtonText?: string;
    /**
     * @default 'SELL'
     * Text for sell order submit button.
     */
    submitSellButtonText?: string;
    /**
     * @default 'Buy'
     * Text for Buy tab label.
     */
    labelFirst?: string;
    /**
     * @default 'Sell'
     * Text for Sell tab label.
     */
    labelSecond?: string;
    orderTypes?: DropdownElem[];
    /**
     *
     */
    width?: number;
}
declare class Order extends React.PureComponent<OrderComponentProps> {
    render(): JSX.Element;
    private getPanels;
    private getPanelsBuy;
    private getPanelsSell;
    private handleChangeTab;
}
export { Order, };
