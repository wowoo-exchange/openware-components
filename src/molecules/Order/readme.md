Order example:

```js

const handleChangePrice = (value) => { console.log(price); }
const onSubmit = (value) => console.log(value);

<Order
  feeBuy={0.0001}
  feeSell={0.0002}
  availableBase={123.10}
  availableQuote={50}
  onSubmit={onSubmit}
  priceMarketBuy={5}
  priceMarketSell={10}
  currentMarketAskPrecision={5}
  currentMarketBidPrecision={4}
  from="btc"
  to="eth"
  orderTypeText={'Order Type'}
  priceText={'Price'}
  amountText={'Amount'}
  handleChangeInputPrice={handleChangePrice}
/>
```
