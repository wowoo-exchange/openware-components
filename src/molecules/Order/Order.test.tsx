import { mount, shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { SinonSpy, spy } from 'sinon';
import { TabPanel } from '../';
import { Order, OrderComponentProps } from './Order';

// tslint:disable:no-magic-numbers

const defaultProps: OrderComponentProps = {
    feeBuy: 10,
    feeSell: 10,
    onSubmit: spy(),
    priceMarketBuy: 5,
    priceMarketSell: 10,
    currentMarketAskPrecision: 4,
    currentMarketBidPrecision: 5,
    availableBase: 200,
    availableQuote: 12,
    from: 'btc',
    to: 'eth',
};

const setup = (props: Partial<OrderComponentProps> = {}) =>
    shallow(<Order {...{ ...defaultProps, ...props }} />);

describe('Order', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = setup();
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render tabs', () => {
        const tabPanel = wrapper.find(TabPanel);
        expect(tabPanel).toHaveLength(2);
    });

    it('button should be disabled', () => {
        // tslint:disable: no-shadowed-variable
        const wrapper = mount(<Order {...{ ...defaultProps}} />);
        wrapper.find('input').at(2).simulate('click');
        expect((defaultProps.onSubmit as SinonSpy).calledOnceWith()).toBeFalsy();
        wrapper.unmount();
    });

    it('should handle submit function', () => {
        // tslint:disable: no-shadowed-variable
        const wrapper = mount(<Order {...{ ...defaultProps}} />);
        wrapper.find('input').at(0).simulate('change', {target: { value: '123' }});
        wrapper.find('input').at(1).simulate('change', {target: { value: '123' }});
        wrapper.find('input').at(2).simulate('click');
        expect((defaultProps.onSubmit as SinonSpy).calledOnceWith()).toBeTruthy();
        wrapper.unmount();
    });
});
