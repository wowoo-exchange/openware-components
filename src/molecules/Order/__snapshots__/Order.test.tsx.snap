// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match snapshot 1`] = `
<div
  className="cr-order cr-order--extended"
>
  <div
    className="cr-order--extended__buy"
  >
    <TabPanel
      fixed={true}
      hideMode="hide"
      onTabChange={[Function]}
      panels={
        Array [
          Object {
            "content": <OrderForm
              available={12}
              currentMarketAskPrecision={4}
              currentMarketBidPrecision={5}
              fee={10}
              from="btc"
              onSubmit={[Function]}
              orderTypes={
                Array [
                  "Limit",
                  "Market",
                ]
              }
              priceMarket={5}
              to="eth"
              type="buy"
            />,
            "label": "Buy",
          },
        ]
      }
    />
  </div>
  <div
    className="cr-order--extended__sell"
  >
    <TabPanel
      fixed={true}
      hideMode="hide"
      onTabChange={[Function]}
      panels={
        Array [
          Object {
            "content": <OrderForm
              available={200}
              currentMarketAskPrecision={4}
              currentMarketBidPrecision={5}
              fee={10}
              from="btc"
              onSubmit={[Function]}
              orderTypes={
                Array [
                  "Limit",
                  "Market",
                ]
              }
              priceMarket={10}
              to="eth"
              type="sell"
            />,
            "label": "Sell",
          },
        ]
      }
    />
  </div>
</div>
`;
